File: reference/CodeExamples/dotNet/CSharpExamples/Code/practises/relations/ShoppingCard.cs
using System.Collections.Generic;

namespace Db4oDoc.Code.Practises.Relations
{
    internal class ShoppingCard
    {
        // #example: Simple 1-n relation. Navigating from the card to the items
        private readonly ICollection<Item> items = new HashSet<Item>();

        public void Add(Item terrain)
        {
            items.Add(terrain);
        }

        public void Remove(Item o)
        {
            items.Remove(o);
        }
        // #end example
    }
}

File: reference/CodeExamples/dotNet/VisualBasicExamples/Code/practises/relations/ShoppingCard.vb
Imports System.Collections.Generic

Namespace Db4oDoc.Code.Practises.Relations
    Friend Class ShoppingCard
        ' #example: Simple 1-n relation. Navigating from the card to the items
        Private ReadOnly items As ICollection(Of Item) = New HashSet(Of Item)()

        Public Sub Add(terrain As Item)
            items.Add(terrain)
        End Sub

        Public Sub Remove(o As Item)
            items.Remove(o)
        End Sub
        ' #end example
    End Class
End Namespace
